File: ruby/medium/range_addition.rb
# @param {Integer} length
# @param {Integer[][]} updates
# @return {Integer[]}
def get_modified_array(length, updates)
    res = Array.new(length).fill(0)
    updates.each do |u|
        i, j, val = u[0], u[1], u[2]
        res[i] += val
        res[j+1] -= val if j < length-1
    end
    sum = 0
    res.each_with_index do |n, i|
        sum += n
        res[i] = sum
    end
    return res
end

File: javascript/medium/rangeAddition.js
/**
 * @param {number} length
 * @param {number[][]} updates
 * @return {number[]}
 */
var getModifiedArray = function(length, updates) {
    const res = new Array(length).fill(0);
    for(let update of updates) {
        const i = update[0], j = update[1], val = update[2];
        res[i] += val;
        if(j < length-1) {
            res[j+1] -= val;    
        }
    }
    for(let i = 0, sum = 0; i < res.length; i++) {
        sum += res[i];
        res[i] = sum;
    }
    return res;
};
